roles/infrastructure/tasks/database.yml

```diff
@@ -0,0 +1,99 @@
+---
+- name: Create PostgreSQL Flexible Server
+  azure.azcollection.azure_rm_resource:
```

Comment:
I tried so many things to get this to be idempotent, but just couldn't. With the other suggestions, this got down to the only non-idempotent thing. It didn't create a new db or anything though. In the past, running the azure_rm_resource_info before on the same object would normally clue into what the API difference is, like caps or param type, but I couldn't spot it this round.

Reply:
Same, initially I thought that it was related to something like https://github.com/ansible/ansible/issues/56356, but I tried a few things without success.